aiohttp/worker.py

```diff
@@ -55,6 +56,9 @@ def run(self) -> None:
             self.loop.run_until_complete(self._task)
         except Exception:
             self.log.exception("Exception in gunicorn worker")
+            self.booted = False
+            self.exit_code = Arbiter.WORKER_BOOT_ERROR
```

Comment:
Hmm, the gunicorn code seems to set `self.booted = True` before it calls `.run()`. This suggests to me that it is about recording whether the process was created, rather than whether the application startup succeeded. This seems to me like it should probably exit 1 instead.

If this was how gunicorn was supposed to work, then I'd expect `booted` to still be `False` and then we would set it to `True` in `_run()` after the setup is complete (line 95).

I'd also assume that you'd want it to retry when failing on startup, as it could be caused by a temporary network failure when connecting to the DB or similar.

Looking at some of the other workers, I'm also noticing gtornado using `alive` and `server_alive`, not sure if something like that is more appropriate:
https://github.com/benoitc/gunicorn/blob/master/gunicorn/workers/gtornado.py#L91-L92

Also wondering if we should actually be using the `AsyncWorker` as our base class:
https://github.com/benoitc/gunicorn/blob/master/gunicorn/workers/base_async.py

Reply:
In fact there is an APP_LOAD_ERROR, which sounds a lot more accurate for this situation:
https://github.com/benoitc/gunicorn/blob/master/gunicorn/arbiter.py#L34

Reply:
I've been thinking about using APP_LOAD_ERROR before, but decided to stay with WORKER_BOOT_ERROR (I didn't know when we can say the app is loaded, but the same problem I have got with a worker). Nevertheless you are probably right about setting `booted` to `False`after it has already been set to `True`, so APP_LOAD_ERROR seems to be correct.

I will take a look at the other workers to have a bit more knowledge.

Reply:
> (I didn't know when we can say the app is loaded, but the same problem I have got with a worker).

When the setup has completed on line 95, the application has been initialised. If we go this way, then I'd suggest adding a `self.started = False` and setting it to `True` after `site.start()` on line 105. Then only set this exit status if `not self.started`, otherwise a `1` is probably an reasonable option (likely indicating that an exception happened in the app's cleanup).